x-pack/plugins/apm/server/lib/anomaly_detection/create_anomaly_detection_jobs.ts

```diff
@@ -78,13 +79,12 @@ async function createAnomalyDetectionJob({
   environment: string;
   indexPatternName?: string | undefined;
 }) {
-  const convertedEnvironmentName = convertToMLIdentifier(environment);
   const randomToken = uuid().substr(-4);
 
   return ml.modules.setup({
     moduleId: ML_MODULE_ID_APM_TRANSACTION,
-    prefix: `${ML_GROUP_NAME_APM}-${convertedEnvironmentName}-${randomToken}-`,
-    groups: [ML_GROUP_NAME_APM, convertedEnvironmentName],
+    prefix: `${ML_GROUP_NAME_APM}-${snakeCase(environment)}-${randomToken}-`,
```

Comment:
I was not aware that `snakeCase` also deburrs and removes non-alphanumeric characters. That's great!
I think it would make sense to call out more explicitly

```ts
// removes all non-alphanumeric characters
const convertedEnvironment = snakeCase(environment);
```